File: src/main/java/tafqeetj/numbers/positive/PositiveQuantitativeNumberNamesFactory.java
package tafqeetj.numbers.positive;

import tafqeetj.numbers.QuantitiveNumberNames;
import tafqeetj.numbers.QuantitativeNumberNamesFactory;

public class PositiveQuantitativeNumberNamesFactory extends QuantitativeNumberNamesFactory {


    private static PositiveQuantitativeNumberNamesFactory instance;

    private PositiveQuantitativeNumberNamesFactory() {
    }

    public synchronized static PositiveQuantitativeNumberNamesFactory getInstance() {
        if (instance == null)
            instance = new PositiveQuantitativeNumberNamesFactory();
        return instance;
    }
}

File: src/main/kotlin/tech/moussa/tafqeetk/numbers/positive/PositiveQuantitativeNumberNamesFactory.kt
package tech.moussa.tafqeetk.numbers.positive

import tech.moussa.tafqeetk.numbers.QuantitativeNumberNamesFactory

class PositiveQuantitativeNumberNamesFactory private constructor(): QuantitativeNumberNamesFactory() {
    companion object {
        private var instance: PositiveQuantitativeNumberNamesFactory? = null

        @Synchronized
        fun getInstance(): PositiveQuantitativeNumberNamesFactory {
            if (PositiveQuantitativeNumberNamesFactory.instance == null) {
                PositiveQuantitativeNumberNamesFactory.instance =
                        PositiveQuantitativeNumberNamesFactory()
            }
            return PositiveQuantitativeNumberNamesFactory.instance!!
        }
    }
}